url8d3c4d17f5_catseye_JaC64_tgz-pJ8-JaC64J8/src/com/dreamfabric/jac64/C64Screen.java

```diff
@@ -0,0 +1,2025 @@
+              // If collission with bg then notice!
+              if ((tmp & 0x100) != 0) {
+                sprBgCol |= smult;
+//              monitor.info("***** Sprite x Bkg collission!");
+              }
+              // If collission with sprite, all colls must
+              // be registered!
+              if ((tmp & 0xff) != smult) {
+                sprCol |= tmp & 0xff;
+//              monitor.info("***** Sprite x Sprite collission: d01e = " +
+//              sprCol + " sprite: " + i + " => " +
+//              smult + " at " + j + "," + vbeam);
+              }
+            }
+          }
+
+          if (SPRITEDEBUG) {
+            if ((sprite.nextByte == 3) && ((j & 4) == 0)) {
+              mem[mpos + j] = 0xff00ff00;
+            }
+            if ((sprite.nextByte == 63) && ((j & 4) == 0)) {
+              mem[mpos + j] = 0xffff0000;
+            }
+
+            if (j == x) {
+              mem[mpos + j] = 0xff000000 + sprite.pointer;
+            }
+          }
+        }
+      }
+    }
+    xPos += 8;
+  }
+
+  public void stop() {
+    motorSound(false);
+    sidChip.stop();
+    audioDriver.shutdown();
+  }
+
+  public void reset() {
+    // Clear a lot of stuff...???
+    initUpdate();
+    sidChip.reset();
+    lastLine = cpu.cycles;
+    nextIOUpdate = cpu.cycles + 47;
+
+    for (int i = 0; i < mem.length; i++) mem[i] = 0;
+    reset = 100;
+
+    sprCol = 0;
+    sprBgCol = 0;
+
+    cia[0].reset();
+    cia[1].reset();
+//  c1541.reset();
+    keyboard.reset();
+    ciaWrites = 0;
+    isrRunning = false;
+
+    motorSound(false);
+    resetInterrupts();
+  }
+
+  public static final int IMG_TOTWIDTH = SC_WIDTH;
+  public static final int IMG_TOTHEIGHT = SC_HEIGHT;
+
+  public Image crtImage;
+
+  // Will be called from the c64canvas class
+  long repaint = 0;
+  public void paint(Graphics g) {
+    if (g == null)
+      return;
+
+    if (image == null) {
+      image = canvas.createImage(IMG_TOTWIDTH, IMG_TOTHEIGHT);
+      g2 = image.getGraphics();
+      g2.setFont(new Font("Monospaced", Font.PLAIN, 11));
+    }
+
+    if (crtImage == null) {
+      crtImage = new BufferedImage(displayWidth, displayHeight,
+          BufferedImage.TYPE_INT_ARGB);
+      Graphics gcrt = crtImage.getGraphics();
+      gcrt.setColor(TRANSPARENT_BLACK);
+      for (int i = 0, n = displayHeight; i < n; i += 2) {
+        gcrt.drawLine(0, i, displayWidth, i);
+      }
+    }
+
+    // Why is there transparency?
+    g2.drawImage(screen, 0, 0,  null);
+
+    if (reset > 0) {
+      g2.setColor(darks[colIndex]);
+      int xp = 44;
+      if (reset < 44) {
+        xp = reset;
+      }
+      g2.drawString("JaC64 " + version + " - Java C64 - www.jac64.com",
+          xp + 1, 9);
+      g2.setColor(lites[colIndex]);
+      g2.drawString("JaC64 " + version + " - Java C64 - www.jac64.com",
+          xp, 8);
+      reset--;
+    } else {
+      String msg = "JaC64 ";
+      if ((message != null) && (message != "")) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.